src/backend/distributed/cdc/cdc_decoder_utils.c

```diff
@@ -351,21 +351,28 @@ CdcPartitionMethodViaCatalog(Oid relationId)
 	TupleDesc tupleDescriptor = RelationGetDescr(pgDistPartition);
 	heap_deform_tuple(partitionTuple, tupleDescriptor, datumArray, isNullArray);
 
-	if (isNullArray[Anum_pg_dist_partition_partmethod - 1])
+	if (isNullArray[Anum_pg_dist_partition_partmethod - 1] ||
+		isNullArray[Anum_pg_dist_partition_repmodel - 1])
 	{
-		/* partition method cannot be NULL, still let's make sure */
+		/*
+		 * partition method and replication model cannot be NULL,
+		 * still let's make sure
+		 */
 		heap_freetuple(partitionTuple);
 		table_close(pgDistPartition, NoLock);
-		return DISTRIBUTE_BY_INVALID;
+		return false;
 	}
 
 	Datum partitionMethodDatum = datumArray[Anum_pg_dist_partition_partmethod - 1];
 	char partitionMethodChar = DatumGetChar(partitionMethodDatum);
 
+	Datum replicationModelDatum = datumArray[Anum_pg_dist_partition_repmodel - 1];
+	char replicationModelChar = DatumGetChar(replicationModelDatum);
+
 	heap_freetuple(partitionTuple);
 	table_close(pgDistPartition, NoLock);
 
-	return partitionMethodChar;
+	return partitionMethodChar == 'n' && replicationModelChar == 't';
```

Comment:
can you use `IsReferenceTableByDistParams`  so that the logic to decide ref table is not scattered?

Reply:
We discussed offline and decided not to use this since this is a separate extension